src/main/scala/com/amazon/deequ/analyzers/Compliance.scala

```diff
@@ -34,8 +35,12 @@ import com.google.common.annotations.VisibleForTesting
   *                         describing what the analysis being done for.
   * @param predicate SQL-predicate to apply per row
   * @param where Additional filter to apply before the analyzer is run.
+  * @param columns List of columns used for predicate - This is needed to run pre condition check!
   */
-case class Compliance(instance: String, predicate: String, where: Option[String] = None)
+case class Compliance(instance: String,
+                      predicate: String,
+                      where: Option[String] = None,
+                      columns: Option[List[String]] = None)
```

Comment:
```suggestion
                      columns: List[String] = List.empty[String])
```

Reply:
Then whole impl. will be a lot of easier

Reply:
Thank You @explicite! 
I missed to see outside the box and followed other param's pattern, this would simplify it.